kbs/src/api_server.rs

```diff
@@ -0,0 +1,317 @@
+            path: request.path().to_string(),
+        })?;
+    let sub_path = request
+        .match_info()
+        .get("sub_path")
+        .ok_or(Error::IllegalAccessedPath {
+            path: request.path().to_string(),
+        })?;
+
+    let end_point = format!("{plugin_name}{sub_path}");
+
+    match plugin_name {
+        #[cfg(feature = "as")]
+        "auth" if request.method() == Method::POST => core
+            .attestation_service
+            .auth(&body)
+            .await
+            .map_err(From::from),
+        #[cfg(feature = "as")]
+        "attest" if request.method() == Method::POST => core
+            .attestation_service
+            .attest(&body, request)
+            .await
+            .map_err(From::from),
+        #[cfg(feature = "as")]
+        "attestation-policy" if request.method() == Method::POST => {
+            core.admin_auth.validate_auth(&request)?;
+
+            core.attestation_service.set_policy(&body).await?;
+
+            Ok(HttpResponse::Ok().finish())
+        }
+        "resource-policy" if request.method() == Method::POST => {
+            core.admin_auth.validate_auth(&request)?;
+
+            core.policy_engine.set_policy(&body).await?;
+
+            Ok(HttpResponse::Ok().finish())
+        }
+        #[cfg(feature = "resource")]
+        "resource" => {
+            if request.method() == Method::GET {
+                // Resource APIs needs to be authorized by the Token and policy
+                let resource_desc =
+                    sub_path
+                        .strip_prefix('/')
+                        .ok_or(Error::IllegalAccessedPath {
+                            path: end_point.clone(),
+                        })?;
+
+                let token = core
+                    .get_attestation_token(&request)
+                    .await
+                    .map_err(|_| Error::TokenNotFound)?;
+
+                let claims = core.token_verifier.verify(token).await?;
+
+                let claim_str = serde_json::to_string(&claims)?;
+                if !core
+                    .policy_engine
+                    .evaluate(resource_desc, &claim_str)
+                    .await?
+                {
+                    return Err(Error::PolicyDeny);
+                };
+
+                let resource_description = ResourceDesc::try_from(resource_desc)?;
+                let resource = core
+                    .resource_storage
+                    .get_secret_resource(resource_description)
+                    .await?;
+
+                let public_key = core.token_verifier.extract_tee_public_key(claims)?;
+                let jwe = jwe(public_key, resource).map_err(|e| Error::JweError { source: e })?;
+
+                let res = serde_json::to_string(&jwe)?;
+
+                Ok(HttpResponse::Ok()
+                    .content_type("application/json")
+                    .body(res))
+            } else if request.method() == Method::POST {
```

Comment:
I wonder if we should introduce a different endpoint for an admin to set resources. In the future we might want to allow an attested client (i.e. the CDH) to set a resource. That would collide with this.

Reply:
Yes. I think we can follow https://github.com/confidential-containers/trustee/pull/514#discussion_r1779304844. The shortage is that old kbs-client will not work fine with new kbs.
interesting idea about an attested client to set resource. Maybe we could talk about it when we meet such case